lib/logstash/inputs/elasticsearch.rb

```diff
@@ -262,22 +268,41 @@ def run(output_queue)
   end
 
   private
-
+  JOB_NAME = "run query"
   def do_run(output_queue)
     # if configured to run a single slice, don't bother spinning up threads
-    return do_run_slice(output_queue) if @slices.nil? || @slices <= 1
+    if @slices.nil? || @slices <= 1
+      return retryable(JOB_NAME) do
+        do_run_slice(output_queue)
+      end
+    end
 
     logger.warn("managed slices for query is very large (#{@slices}); consider reducing") if @slices > 8
 
     pipeline_id = execution_context&.pipeline_id || 'main'
     @slices.times.map do |slice_id|
       Thread.new do
         LogStash::Util::set_thread_name("[#{pipeline_id}]|input|elasticsearch|slice_#{slice_id}")
-        do_run_slice(output_queue, slice_id)
+        retryable(JOB_NAME) do
```

Comment:
QQ: with `slices` provided case, it seems we will face consistency issue here. Eg.: if we are dividing the work into 5 slices and first 3 slices fails with retry, then 2 succeeds. Maybe my ideation would not be a valid but IMHO, retrying seems not an ideal case for slices.

Reply:
@mashhurs Thanks for raising the concern about data integrity. I changed the PR to discard the whole batch if one of the slices fails. Please review it.